src/lib/dnssd/minimal_mdns/Server.cpp

```diff
@@ -48,6 +52,48 @@ class ShutdownOnError
     ServerBase * mServer;
 };
 
+class ListenSocketDelegate : public ServerBase::BroadcastSendDelegate
+{
+public:
+    chip::Inet::UDPEndPoint * Accept(ServerBase::EndpointInfo * info) override { return info->listen_udp; }
+};
+
+class QuerySocketDelegate : public ServerBase::BroadcastSendDelegate
+{
+public:
+    chip::Inet::UDPEndPoint * Accept(ServerBase::EndpointInfo * info) override { return info->query_udp; }
+};
+
+class InterfaceTypeValidateDelegate : public ServerBase::BroadcastSendDelegate
+{
+public:
+    InterfaceTypeValidateDelegate(chip::Inet::InterfaceId interface, chip::Inet::IPAddressType type,
```

Comment:
It's not clear by the name and without comments what this class is for and why it's named like this

Reply:
Added comments.